File: Data Structures & Algorithms/three-integer-sum/submission-8.py
class Solution:
    def threeSum(self, nums: List[int]) -> List[List[int]]:
        nums.sort()
        ans = []
        for i, n in enumerate(nums):
            if i > 0 and n == nums[i - 1]:
                continue
            target = 0 - n
            l, r = i + 1, len(nums) - 1
            while l < r:
                s = nums[l] + nums[r]
                if s > target:
                    r -= 1
                elif s < target:
                    l += 1
                else:
                    ans.append([n, nums[l], nums[r]])
                    while l < r and nums[r] == nums[r - 1]:
                        r -= 1
                    r -= 1
                    l += 1
        return ans

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-8.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_set<int> items(nums.begin(), nums.end());
        int ans = 0;
        for (const int n : nums) {
            int target = n;
            if (items.find(target - 1) != items.end()) continue;
            size_t curr = 1;
            while (items.find(target + 1) != items.end()) {
                target++;
                curr++;
            }
            ans = std::max<int>(curr, ans);
        }
        return ans;
    }
};
